### PR TITLE
refactor: Clean up agent preset error handling and add comprehensive tests

### PR DESCRIPTION
**Pull request by daryllimyt**
## Summary
- Simplified error handling in agent preset routes and service methods
- Fixed `AgentPresetUpdate` schema to make all fields optional
- Added comprehensive test suite for `AgentPresetService` (27 tests)

## Changes
- **Service**: Removed redundant error handling and let exceptions propagate naturally
- **Router**: Simplified route handlers by relying on service methods to raise appropriate exceptions
- **Schema**: Fixed `AgentPresetUpdate` to properly inherit optional fields instead of required ones from base
- **Tests**: Added full test coverage for CRUD operations, validation, config resolution, and edge cases

## Test Coverage
All 27 tests passing, covering:
- CRUD operations (create, read, update, delete)
- Slug validation and normalization
- Action validation against registry
- Agent config resolution
- Error handling for edge cases

---
## Summary by cubic
Streamlined agent preset error handling and fixed the update schema to support true partial updates, then added a comprehensive test suite. This reduces API boilerplate, improves 404 messages, and aligns frontend types with the API.

- **Refactors**
  - Simplified router and service error handling: router returns clear 404s; service get-by-id/slug return None; update/delete accept model instances; centralized slug and action validation.
  - Fixed AgentPresetUpdate to allow partial updates; regenerated frontend schemas and types to reflect optional/nullable fields (no required model_name/model_provider).

- **Tests**
  - Added 27 unit tests covering CRUD, slug normalization, action validation, agent config resolution, and error paths (all passing).

<sup>Written for commit 4bbdd7a0adff914f40475cddb9d827bd63c8741d. Summary will update automatically on new commits.</sup>

### CODE REVIEW

File: tracecat/agent/preset/service.py [hunk 1]
```diff
@@ -92,10 +81,9 @@ async def _validate_actions(self, actions: list[str]) -> None:
             )
 
     async def update_preset(
-        self, preset_id: uuid.UUID, params: AgentPresetUpdate
+        self, preset: AgentPreset, params: AgentPresetUpdate
```
[the remark this answers is not in the document]
Comment (daryllimyt) [reply tracecat/agent/preset/service.py:84]: we have updated callers

**Comment by daryllimyt** [comment]
@cubic-dev-ai rereview

**Comment by daryllimyt** [comment]
@codex rereview
